src/test/java/edu/kis/vh/nursery/RhymersJUnitTest.java

```diff
@@ -1,82 +1,147 @@
+    public void testPeekabooList() {
+        DefaultCountingOutRhymer rhymer = new DefaultCountingOutRhymer(new IntLinkedList());
+        final int EMPTY_STACK_VALUE = -1;
+
+        Assert.assertEquals(EMPTY_STACK_VALUE, rhymer.peekaboo());
+
+        final int testValue = 4;
+        rhymer.countIn(testValue);
 
-	@Test
-	public void testPeekaboo() {
-		DefaultCountingOutRhymer rhymer = new DefaultCountingOutRhymer();
-		final int EMPTY_STACK_VALUE = -1;
+        Assert.assertEquals(testValue, rhymer.peekaboo());
+        Assert.assertEquals(testValue, rhymer.peekaboo());
+    }
 
-		Assert.assertEquals(EMPTY_STACK_VALUE, rhymer.peekaboo());
+    @Test
+    public void testPeekabooArray() {
+        DefaultCountingOutRhymer rhymer = new DefaultCountingOutRhymer(new IntArrayStack());
+        final int EMPTY_STACK_VALUE = -1;
 
-		final int testValue = 4;
-		rhymer.countIn(testValue);
+        int result = rhymer.peekaboo();
+        Assert.assertEquals(EMPTY_STACK_VALUE, result);
 
-		Assert.assertEquals(testValue, rhymer.peekaboo());
-		Assert.assertEquals(testValue, rhymer.peekaboo());
-	}
+        int testValue = 4;
+        rhymer.countIn(testValue);
 
-	@Test
-	public void testCountOut() {
-		DefaultCountingOutRhymer rhymer = new DefaultCountingOutRhymer();
-		final int EMPTY_STACK_VALUE = -1;
+        Assert.assertEquals(testValue, rhymer.peekaboo());
+        Assert.assertEquals(testValue, rhymer.peekaboo());
+    }
 
-		int result = rhymer.countOut();
-		Assert.assertEquals(EMPTY_STACK_VALUE, result);
+    @Test
+    public void testCountOutList() {
+        DefaultCountingOutRhymer rhymer = new DefaultCountingOutRhymer(new IntLinkedList());
+        final int EMPTY_STACK_VALUE = 0;
 
-		final int testValue = 4;
-		rhymer.countIn(testValue);
+        int result = rhymer.countOut();
+        Assert.assertEquals(EMPTY_STACK_VALUE, result);
 
-		result = rhymer.countOut();
-		Assert.assertEquals(testValue, result);
-		result = rhymer.countOut();
-		Assert.assertEquals(EMPTY_STACK_VALUE, result);
-	}
+        final int testValue = 4;
+        rhymer.countIn(testValue);
+
+        Assert.assertEquals(testValue, rhymer.countOut());
+        Assert.assertEquals(EMPTY_STACK_VALUE, rhymer.countOut());
+    }
+
+    @Test
+    public void testCountOutArray() {
+        DefaultCountingOutRhymer rhymer = new DefaultCountingOutRhymer(new IntArrayStack());
+        final int EMPTY_STACK_VALUE = 0;
+
+        int result = rhymer.countOut();
+        Assert.assertEquals(EMPTY_STACK_VALUE, result);
+
+        int testValue = 4;
+        rhymer.countIn(testValue);
+
+        Assert.assertEquals(testValue, rhymer.countOut());
+        Assert.assertEquals(EMPTY_STACK_VALUE, rhymer.countOut());
+    }
 
     @Test
     public void testCountOutFIFORhymer() {
-        FIFORhymer rhymer = new FIFORhymer();
-        final int EMPTY_STACK_VALUE = -1;
+        FIFORhymer rhymer = new FIFORhymer(new IntArrayStack());
+        final int EMPTY_STACK_VALUE = 0;
```

Comment:
The original constant should be used (it would be best to use a common constant in a stacks interface).